improver/utilities/compare.py

```diff
@@ -242,8 +291,10 @@ def compare_attributes(name, actual_ds, desired_ds, reporter):
     desired_attrs = set(desired_ds.ncattrs())
     # ignore history attribute - this often contain datestamps and other
     # overly specific details
-    actual_attrs.discard("history")
-    desired_attrs.discard("history")
+    if ignored_attributes is not None:
```

Comment:
The default case of `ignored_attributes=None` here will cause no attributes to be ignored. The `compare` function for acceptance test use overrides that with `IGNORED_ATTRIBUTES` from `improver_tests/acceptance/acceptance.py`.
My impression from reading this diff is that the `compare` CLI will now behave differently to the acceptance test usage, since it doesn't override the default. Previously both CLI and acceptance test usages had the same ignore of the history attribute.

Is this behaviour change intended?

Reply:
It was not, I didn't know the CLI existed. I'm not sure what the use cases are for the CLI (?), but I have added an optional argument to access the ignored_attributes option. I have not set a default value, though this could be set to "history" if this CLI is currently in use and that would be preferable.

I have added an additional acceptance test to cover this.